examples/src/main/java/kafka/examples/TransactionalClientDemo.java

```diff
@@ -0,0 +1,148 @@
+import static org.apache.kafka.clients.consumer.ConsumerConfig.*;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG;
+import static org.apache.kafka.clients.producer.ProducerConfig.*;
+
+public class TransactionalClientDemo {
+
+    private static final String CONSUMER_GROUP_ID = "my-group-id";
+    private static final String OUTPUT_TOPIC = "output";
+    private static final String INPUT_TOPIC = "input";
+    private static KafkaConsumer<String, String> consumer;
+    private static KafkaProducer<String, String> producer;
+
+    public static void main(String[] args) {
+        initializeApplication();
+
+        boolean isRunning = true;
+        // Continuously poll for records
+        while(isRunning) {
+            try {
+                try {
+                    // Poll records from Kafka for a timeout of 60 seconds
+                    ConsumerRecords<String, String> records = consumer.poll(ofSeconds(60));
+
+                    // Process records to generate word count map
+                    Map<String, Integer> wordCountMap = records.records(new TopicPartition(INPUT_TOPIC, 0))
+                            .stream()
+                            .flatMap(record -> Stream.of(record.value().split(" ")))
+                            .map(word -> new AbstractMap.SimpleEntry<>(word, 1))
+                            .collect(Collectors.toMap(
+                                    AbstractMap.SimpleEntry::getKey,
+                                    AbstractMap.SimpleEntry::getValue,
+                                    (v1, v2) -> v1 + v2
+                            ));
+
+                    // Begin transaction
+                    producer.beginTransaction();
+
+                    // Produce word count results to output topic
+                    wordCountMap.forEach((key, value) ->
+                            producer.send(new ProducerRecord<>(OUTPUT_TOPIC, key, value.toString())));
+
+                    // Determine offsets to commit
+                    Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = new HashMap<>();
+                    for (TopicPartition partition : records.partitions()) {
+                        List<ConsumerRecord<String, String>> partitionedRecords = records.records(partition);
+                        long offset = partitionedRecords.get(partitionedRecords.size() - 1).offset();
+                        offsetsToCommit.put(partition, new OffsetAndMetadata(offset + 1));
+                    }
+
+                    // Send offsets to transaction for atomic commit
+                    producer.sendOffsetsToTransaction(offsetsToCommit, CONSUMER_GROUP_ID);
+
+                    // Commit transaction
+                    producer.commitTransaction();
+                } catch (AbortableTransactionException e) {
+                    // Abortable Exception: Handle Kafka exception by aborting transaction. Abortable Exception should never be thrown.
```

Comment:
"Abortable Exception should never be thrown."  -- is the desired message that the abortTransaction never throws abortable exceptions?